### ISSUE TITLE
naming/lisp-case false positive for gen-class method names

### ISSUE THREAD

**Issue by lvh** [NoahTheDuke/splint#37]
### Summary
`splint` 1.21.0 reports `naming/lisp-case` warnings for method names that must remain hyphen-prefixed when using `gen-class`. This is a false positive because renaming the function to kebab-case breaks the generated class (Java interop requires the exact method name).

### Repro
```
(ns demo.gen-class
  (:gen-class
   :implements [java.util.concurrent.Callable]))

(defn -handleRequest
  [_]
  :ok)
```

Run:
```
clojure -Sdeps '{:deps {io.github.noahtheduke/splint {:mvn/version "1.21.0"}}}' \
  -M -m noahtheduke.splint splint-repro
```

### Actual
```
splint-repro/gen_class.clj:5:1 [naming/lisp-case] - Prefer kebab-case over other cases for top-level definitions.
```

### Expected
No naming warning. `gen-class` forces the `-handleRequest` name to satisfy the Java interface (`Callable` in this demo, `RequestStreamHandler` in real code).

### Notes
A minimal repro lives here: https://github.com/latacora/libraries/tree/feature/sms-router/splint-repro
